utility/src/main/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImpl.kt

```diff
@@ -70,4 +76,33 @@ class PerformanceMetricsAssessorImpl @Inject constructor(
       else -> HIGH_MEMORY_TIER
     }
   }
+
+  override fun getCurrentCpuUsageParameters(
+    currentApplicationState: ApplicationState
+  ): CpuUsageParameters {
+    return CpuUsageParameters.newBuilder().apply {
+      cpuTime = Process.getElapsedCpuTime()
+      processTime = oppiaClock.getCurrentTimeMs()
+      applicationState = currentApplicationState
+      numberOfActiveCores = getNumberOfCores()
+    }.build()
+  }
+
+  override fun getRelativeCpuUsage(
+    cpuUsageAtStartOfTimeWindow: CpuUsageParameters,
+    cpuUsageAtEndOfTimeWindow: CpuUsageParameters
+  ): Double {
+    val deltaCpuTimeMs = cpuUsageAtEndOfTimeWindow.cpuTime - cpuUsageAtStartOfTimeWindow.cpuTime
+    val deltaProcessTimeMs =
+      (cpuUsageAtEndOfTimeWindow.processTime * cpuUsageAtEndOfTimeWindow.numberOfActiveCores) -
+        (cpuUsageAtStartOfTimeWindow.processTime * cpuUsageAtStartOfTimeWindow.numberOfActiveCores)
+    return (100 * (deltaCpuTimeMs.toDouble() / deltaProcessTimeMs.toDouble()))
+  }
+
+  private fun getNumberOfCores(): Int {
+    val numOfCores = File("/sys/devices/system/cpu/").listFiles()?.filter {
```

Comment:
Why not use ``availableProcessors()``, instead?

Reply:
done.